payments-ui-core/src/main/java/com/stripe/android/ui/core/forms/resources/LpmRepository.kt

```diff
@@ -122,6 +145,8 @@ class LpmRepository constructor(
             // we will use the LPM on disk if found
             val lpmsNotParsedFromServerSpec = expectedLpms
                 .filter { !lpmInitialFormData.containsKey(it) }
+                .filter { supportsPaymentMethod(it) }
```

Comment:
We need to start filtering here to, as we already do for specs coming from the backend. Otherwise, UPI would be available immediately.